source/common/common.c

```diff
@@ -22,6 +24,41 @@ int current_unix_timestamp(void) {
 	ularge.HighPart = file_time.dwHighDateTime;
 	return (long)((ularge.QuadPart - 116444736000000000) / 10000000L);
 }
+
+/*!
+ * @brief Sleep for the given number of seconds.
+ * @param seconds DWORD value representing the number of seconds to sleep.
+ * @remark This was implemented so that extended sleep times can be used (beyond the
+ *         49 day limit imposed by Sleep()). With this implementation it's possible
+ *         to wait for up to 3.1415926 metric buttloads of seconds.
+ *         NtDelayExecution takes a parameter that is measured in 100-ns units, and
+ *         negative values delay the thread relative to the current time. This is
+ *         a 63-bit value (1 bit for sign), which means it should be possible to wait
+ *         for approximately 29,227 years.
+ */

```

Comment:
Not to bikeshed (too late?), but you can do this a little more simply and not have to worry about backward-compat with some extra Sleep iterations instead: https://gist.github.com/bcook-r7/75a4ef47ed78f417eb12